app/src/main/java/com/github/se/studybuddies/MainActivity.kt

```diff
@@ -139,9 +141,14 @@ class MainActivity : ComponentActivity() {
             }
             composable(Route.CALENDAR) {
               ifNotNull(remember { auth.currentUser }) { _ ->
-                val calendarViewModel = remember { CalendarViewModel() }
-                CalendarApp(calendarViewModel, navigationActions)
-                Log.d("MyPrint", "Successfully navigated to GroupsHome")
+                val calendarViewModel = remember {
+                  auth.currentUser?.let { it1 -> CalendarViewModel(it1.uid) }
+                }
+                if (calendarViewModel != null) {
+                  CalendarApp(calendarViewModel, navigationActions)
+                }
```

Comment:
Careful you introduced a big jetpack compose mistake in these few lines! Can you spot it?

<details>
  <summary>Spoiler warning</summary>
  
  How do you create [view models with dependencies in Jetpack compose](https://developer.android.com/topic/libraries/architecture/viewmodel/viewmodel-factories#jetpack-compose)?
  
</details>

Reply:
I used a factory thank you for your comment and the useful link